### PR TITLE
makeDocs script improvements

### PR DESCRIPTION
**Pull request by multimeric**
**Background**
I'm hoping to get automated CLI documentation into the Singularity docs. Refer to the discussion here: https://github.com/sylabs/singularity-userdocs/issues/73. The first step involved in this is getting Singularity to generate ReST docs that can be consumed by the Sphinx builder in the userdocs repo. I've looked at generating them from that repo, but decided it will be much simpler and cleaner to adapt the existing makeDocs script we have here

**Description of the Pull Request (PR):**
- Convert the makeDocs script to a cobra command
- Allow production of markdown and ReST docs.

**This fixes or addresses the following GitHub issues:**
- This is a step towards https://github.com/sylabs/singularity-userdocs/issues/73

**Before submitting a PR, make sure you have done the following:**

- [x] Read the [Guidelines for Contributing](../CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [x] Added changes to the [CHANGELOG](../CHANGELOG.md) if necessary according to the [Contribution Guidelines](../CONTRIBUTING.md)
- [ ] Added tests to validate this PR and tested this PR locally with a `make testall`
- [x] Based this PR against the appropriate branch according to the [Contribution Guidelines](../CONTRIBUTING.md)
- [x] Added myself as a contributor to the [Contributors File](../CONTRIBUTORS.md)

### CODE REVIEW

**Comment by multimeric** [comment]
I'm interested in the devs thoughts on:

1) Is the new makeDocs CLI okay (with `the --dir` flag, and no longer defaulting to the man directory)
2) If, where and how I should add tests for this

File: cmd/docs/docs.go
Comment (WestleyK) [inline cmd/docs/docs.go]: 2019?

```suggestion
// Copyright (c) 2019, Sylabs Inc. All rights reserved.
```

Comment (jmstover) [reply cmd/docs/docs.go]: I agree with the proposed date change here by @WestleyK .

**Review by jmstover — CHANGES_REQUESTED** [review]
If you can change the copyright date, this looks good to me. Ran it, and output all the different formats.

**Comment by multimeric** [comment]
Does anyone think I need to add tests for this before it gets merged?

**Comment by jmstover** [comment]
For this... I personally don't think so. This is really pretty self contained.

On a side note, CircleCI failed because of a resource issue.... but I don't have permissions to restart it. If you do a new commit, it should trigger a re-run.

**Comment by jmstover** [comment]
@TMiguelT ... Thanks. This looks good. Just waiting on the CI to finish up cleanly.

**Review by jmstover — APPROVED** [review]

File: cmd/docs/docs.go
Comment (jchavezlavalle) [inline cmd/docs/docs.go]: This error warning should be updated to ``Failed to create the ReST format of the docs (or ReST docs) for Singularity``

File: cmd/docs/docs.go
Comment (jchavezlavalle) [inline cmd/docs/docs.go]: if this works on all man pages then this warning should be in plural: Failed to create **man pages** for Singularity

**Review by jchavezlavalle — CHANGES_REQUESTED** [review]
Added some comments @TMiguelT thank you!

**Review by jscook2345 — APPROVED** [review]
I approve of this PR, but I will give some feedback on what I may have done differently. Don't feel like you have to make these changes.

I know `ReST` is short for restructured text, but I can't help but think of `REST`. Maybe `RST` would be a better option?

I would of broke each case statement into it's own function for testing purposes. You could than add unit tests (or I can help add them) for each function.

Thanks for your contribution! It's much appreciated!

**Comment by multimeric** [comment]
Okay, I've fixed the issues that have been mentioned:

- Docs/pages pluralised because in all cases it generates multiple files
- ReST changed to RST for clarity in all cases
- Documentation generation has been split into separate functions
- Fixed the log where I printed "markdown" instead of RST/ReST

In addition, I re-added the directory access check that was in the original script. I suspect that Cobra would tell us if it didn't have access but this might be a more user friendly error

File: cmd/docs/docs.go
Comment (jmstover) [inline cmd/docs/docs.go]: Can you change `out_dir` to be `outDir`

```
 CHECK go lint
/go/src/github.com/sylabs/singularity/cmd/docs/docs.go:22:19: don't use underscores in Go names; func parameter out_dir should be outDir
```

**Review by jmstover — CHANGES_REQUESTED** [review]

File: cmd/docs/docs.go
Comment (multimeric) [reply cmd/docs/docs.go]: Done!

**Review by jmstover — APPROVED** [review]

**Review by jchavezlavalle — APPROVED** [review]
lgtm, thanks @TMiguelT !

**Comment by jmstover** [comment]
Thanks @TMiguelT

**Pull request merged**
